src/Controls/tests/DeviceTests/Elements/Image/ImageTests.Android.cs

```diff
@@ -37,6 +39,45 @@ public async Task CachingDisabledForImagesLoadViaInputStreams()
 				await bitmapDrawable1.Bitmap.AssertNotEqualAsync(bitmapDrawable2.Bitmap);
 			});
 		}
+
+		[Fact]
+		public async Task ImageSetFromStreamRenders()
+		{
+			SetupBuilder();
+			var layout = new VerticalStackLayout()
+			{
+				HeightRequest = 100,
```

Comment:
@mattleibow @jstedfast I modified this test to just set the height/width because I don't think the remeasuring of the layout is relevant to the test here. 

I also don't think this test completely accurately embodies the original issue. I rolled back the changes from this PR and this test still passes without throwing a UIThread exception

Reply:
I haven't been able to reproduce the original issue outside the sample repository. 

I started a branch here to attempt reproducing
https://github.com/dotnet/maui/tree/test_for_Issue14052 but haven't had any luck yet. 

I did test this PR directly against the repro and was able to reproduce the crash and then verified this PR fixes the crash.  

This fix has been on NET8 since preview5 so I feel alright to merge this Backport

Reply:
https://github.com/dotnet/maui/issues/17465

Reply:
D'oh, this Width/HeightRequest change makes total sense. I was assuming the issue was that the screenshot was happening before layout had finished or something and so the width/height of the element wasn't yet known (hence the -1 values).

Reply:
FWIW, I think the test only ever randomly failed - I don't think it failed every run.